contrib/postgres-cmake/CMakeLists.txt

```diff
@@ -79,6 +79,25 @@ if (OS_DARWIN OR OS_FREEBSD OR USE_MUSL)
     target_compile_definitions(_libpq PRIVATE -DSTRERROR_R_INT=1)
 endif()
 
+# Mac 15.4 / Xcode 16.3 started defining srtchrnul
+# Could be cleaned up after resolved in upstream https://www.postgresql.org/message-id/flat/385134.1743523038%40sss.pgh.pa.us
+if (OS_DARWIN AND NOT USE_MUSL)
```

Comment:
And let's move `SDK_VERSION` where we define `OS_DARWIN`, and here only check it against 15.4